src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1118,21 +1118,21 @@ private void EmitSwitchDispatch(BoundSwitchDispatch dispatch)
             EmitSwitchHeader(
                 dispatch.Expression,
                 dispatch.Cases.Select(p => new KeyValuePair<ConstantValue, object>(p.value, p.label)).ToArray(),
-                dispatch.DefaultLabel);
+                dispatch.DefaultLabel,
+                dispatch.LengthBasedStringSwitchDataOpt);
         }
 
         private void EmitSwitchHeader(
             BoundExpression expression,
             KeyValuePair<ConstantValue, object>[] switchCaseLabels,
-            LabelSymbol fallThroughLabel)
+            LabelSymbol fallThroughLabel,
+            LengthBasedStringSwitchData lengthBasedSwitchStringJumpTableOpt)
         {
             Debug.Assert(expression.ConstantValue == null);
             Debug.Assert((object)expression.Type != null &&
                 (expression.Type.IsValidV6SwitchGoverningType() || expression.Type.IsSpanOrReadOnlySpanChar()));
             Debug.Assert(switchCaseLabels.Length > 0);
 
-            Debug.Assert(switchCaseLabels != null);
```

Comment:
>Debug.Assert(switchCaseLabels != null); [](http://example.com/codeflow?start=12&length=39)

Perhaps assert that either `switchCaseLabels` or `lengthBasedSwitchStringJumpTableOpt` is not null?